lib/blockstore/file.js

```diff
@@ -0,0 +1,502 @@
+/*!
+ * blockstore/file.js - file blockstore for bcoin
+ * Copyright (c) 2019, Braydon Fuller (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const {isAbsolute, resolve, join} = require('path');
+const bdb = require('bdb');
+const assert = require('bsert');
+const fs = require('bfile');
+const bio = require('bufio');
+const Network = require('../protocol/network');
+const Block = require('../primitives/block');
+const AbstractBlockStore = require('./abstract');
+const {BlockRecord, FileRecord} = require('./records');
+const layout = require('./layout');
+const {types, prefixes} = require('./common');
+
+/**
+ * File Block Store
+ *
+ * @alias module:blockstore:FileBlockStore
+ * @abstract
+ */
+
+class FileBlockStore extends AbstractBlockStore {
+  /**
+   * Create a blockstore that stores blocks in files.
+   * @constructor
+   */
+
+  constructor(options) {
+    super();
+
+    assert(isAbsolute(options.location), 'Location not absolute.');
+
+    this.location = options.location;
+    this.indexLocation = resolve(this.location, './index');
+
+    this.db = bdb.create({
+      location: this.indexLocation,
+      cacheSize: options.cacheSize,
+      compression: false
+    });
+
+    this.maxFileLength = options.maxFileLength || 128 * 1024 * 1024;
```

Comment:
Should we `assert` for a max size here?

Reply:
Can check that it's as number.

Reply:
Sounds good, we should do that `assert` for all the options I think.